### PR TITLE
Network Traffic Metrics

### PR DESCRIPTION
**Pull request by Z-Kris**
Adds support for tracking various network level metrics between the different handlers.
Information tracked includes:
- Timestamps when it was recorded
- The number of connection requests established via Netty
- For Login, JS5 and Game handlers:
  - IP Address
  - Active connections (at the time of the snapshot)
  - Incoming packets (count of each, as well as the sum of all the payloads)
  - Outgoing packets (count of each, as well as the sum of all the payloads)
  - Reasons for channel disconnecting/being rejected
- Login Blocks
  - IP Address
  - Detailed breakdown of most of the information in the login packet

Below is an example of the output with the built in writer class, in which I downloaded a full cache via JS5 and played the game for a few minutes. Additionally, tried to login from the native and java client, failed some, succeeded others.
<details>
  <summary>Network Traffic Snapshot Spoiler</summary>
  
```
Network Traffic Snapshot
  Snapshot started on: Thursday, September 26, 2024 at 5:35:15 PM Greenwich Mean Time
  Snapshot ended on: Thursday, September 26, 2024 at 5:42:33 PM Greenwich Mean Time
  Elapsed duration: 7m 18.128s
  Connection requests: 8

Login Channel Snapshot
  Login snapshot started on: Thursday, September 26, 2024 at 5:35:15 PM Greenwich Mean Time
  Login snapshot ended on: Thursday, September 26, 2024 at 5:42:33 PM Greenwich Mean Time
  Login elapsed duration: 7m 18.149s
  Login INet Address Metrics
    INet Address: /127.0.0.1
      Outgoing packet: DISALLOWED_BY_SCRIPT, count: 2, payload sum: 129 bytes
      Outgoing packet: OK, count: 2, payload sum: 68 bytes
      Outgoing packet: SUCCESSFUL, count: 8, payload sum: 40 bytes
      Disconnection reason: GAME_LOGIN_FAIL_1, count: 1

JS5 Channel Snapshot
  JS5 snapshot started on: Thursday, September 26, 2024 at 5:35:15 PM Greenwich Mean Time
  JS5 snapshot ended on: Thursday, September 26, 2024 at 5:42:33 PM Greenwich Mean Time
  JS5 elapsed duration: 7m 18.152s
  JS5 INet Address Metrics
    INet Address: /127.0.0.1
      Incoming packet: PREFETCH_REQUEST, count: 100,906, payload sum: 302,718 bytes
      Incoming packet: URGENT_REQUEST, count: 8,803, payload sum: 26,409 bytes
      Incoming packet: PRIORITY_CHANGE_HIGH, count: 2, payload sum: 6 bytes
      Incoming packet: PRIORITY_CHANGE_LOW, count: 2, payload sum: 6 bytes
      Outgoing packet: JS5_GROUP_RESPONSE, count: 109,709, payload sum: 147,108,302 bytes
      Disconnection reason: IDLE, count: 3

Game Channel Snapshot
  Game snapshot started on: Thursday, September 26, 2024 at 5:35:15 PM Greenwich Mean Time
  Game snapshot ended on: Thursday, September 26, 2024 at 5:42:33 PM Greenwich Mean Time
  Game elapsed duration: 7m 18.147s
  Game INet Address Metrics
    INet Address: /127.0.0.1
      Active connections: 1
      Incoming packet: EVENT_MOUSE_MOVE, count: 256, payload sum: 7,009 bytes
      Incoming packet: EVENT_KEYBOARD, count: 301, payload sum: 1,208 bytes
      Incoming packet: EVENT_MOUSE_CLICK, count: 178, payload sum: 1,068 bytes
      Incoming packet: EVENT_CAMERA_POSITION, count: 126, payload sum: 504 bytes
      Incoming packet: MOVE_GAMECLICK, count: 52, payload sum: 260 bytes
      Incoming packet: OPLOC1, count: 31, payload sum: 217 bytes
      Incoming packet: IF_BUTTON2, count: 26, payload sum: 208 bytes
      Incoming packet: EVENT_NATIVE_MOUSE_MOVE, count: 1, payload sum: 190 bytes
      Incoming packet: EVENT_MOUSE_SCROLL, count: 72, payload sum: 144 bytes
      Incoming packet: IF_BUTTON1, count: 17, payload sum: 136 bytes
      Incoming packet: CLIENT_CHEAT, count: 7, payload sum: 61 bytes
      Incoming packet: OPLOC2, count: 8, payload sum: 56 bytes
      Incoming packet: WINDOW_STATUS, count: 9, payload sum: 45 bytes
      Incoming packet: CONNECTION_TELEMETRY, count: 2, payload sum: 28 bytes
      Incoming packet: TELEPORT, count: 3, payload sum: 27 bytes
      Incoming packet: MOVE_MINIMAPCLICK, count: 1, payload sum: 18 bytes
      Incoming packet: EVENT_APPLET_FOCUS, count: 15, payload sum: 15 bytes
      Incoming packet: IF_BUTTON3, count: 1, payload sum: 8 bytes
      Incoming packet: DETECT_MODIFIED_CLIENT, count: 1, payload sum: 4 bytes
      Incoming packet: SOUND_JINGLEEND, count: 1, payload sum: 4 bytes
      Incoming packet: MEMBERSHIP_PROMOTION_ELIGIBILITY, count: 1, payload sum: 2 bytes
      Incoming packet: NO_TIMEOUT, count: 128, payload sum: 0 bytes
      Incoming packet: MAP_BUILD_COMPLETE, count: 8, payload sum: 0 bytes
      Incoming packet: CLOSE_MODAL, count: 2, payload sum: 0 bytes
      Outgoing packet: NPC_INFO_LARGE, count: 522, payload sum: 55,130 bytes
      Outgoing packet: UPDATE_ZONE_PARTIAL_ENCLOSED, count: 375, payload sum: 10,417 bytes
      Outgoing packet: REBUILD_NORMAL, count: 7, payload sum: 10,008 bytes
      Outgoing packet: UPDATE_INV_FULL, count: 22, payload sum: 9,640 bytes
      Outgoing packet: UPDATE_ZONE_FULL_FOLLOWS, count: 2,583, payload sum: 7,749 bytes
      Outgoing packet: VARP_LARGE, count: 1,088, payload sum: 6,528 bytes
      Outgoing packet: SET_ACTIVE_WORLD, count: 1,043, payload sum: 4,172 bytes
      Outgoing packet: OBJ_ADD, count: 263, payload sum: 3,682 bytes
      Outgoing packet: PLAYER_INFO, count: 522, payload sum: 2,971 bytes
      Outgoing packet: MESSAGE_GAME, count: 41, payload sum: 2,593 bytes
      Outgoing packet: VARP_SMALL, count: 783, payload sum: 2,349 bytes
      Outgoing packet: RUNCLIENTSCRIPT, count: 106, payload sum: 2,099 bytes
      Outgoing packet: SET_NPC_UPDATE_ORIGIN, count: 1,044, payload sum: 2,088 bytes
      Outgoing packet: SYNTH_SOUND, count: 290, payload sum: 1,450 bytes
      Outgoing packet: IF_SETEVENTS, count: 92, payload sum: 1,104 bytes
      Outgoing packet: IF_SETTEXT, count: 60, payload sum: 921 bytes
      Outgoing packet: UPDATE_ZONE_PARTIAL_FOLLOWS, count: 228, payload sum: 684 bytes
      Outgoing packet: UPDATE_RUNENERGY, count: 334, payload sum: 668 bytes
      Outgoing packet: IF_OPENSUB, count: 87, payload sum: 609 bytes
      Outgoing packet: UPDATE_STAT, count: 80, payload sum: 560 bytes
      Outgoing packet: WORLDENTITY_INFO, count: 521, payload sum: 521 bytes
      Outgoing packet: UPDATE_INV_PARTIAL, count: 29, payload sum: 425 bytes
      Outgoing packet: SET_MAP_FLAG, count: 150, payload sum: 300 bytes
      Outgoing packet: LOC_ADD_CHANGE, count: 54, payload sum: 270 bytes
      Outgoing packet: MIDI_SONG, count: 22, payload sum: 220 bytes
      Outgoing packet: IF_CLOSESUB, count: 40, payload sum: 160 bytes
      Outgoing packet: LOC_DEL, count: 54, payload sum: 108 bytes
      Outgoing packet: UPDATE_RUNWEIGHT, count: 44, payload sum: 88 bytes
      Outgoing packet: MINIMAP_TOGGLE, count: 84, payload sum: 84 bytes
      Outgoing packet: SET_PLAYER_OP, count: 4, payload sum: 44 bytes
      Outgoing packet: IF_SETHIDE, count: 3, payload sum: 15 bytes
      Outgoing packet: IF_OPENTOP, count: 4, payload sum: 8 bytes
      Outgoing packet: CHAT_FILTER_SETTINGS, count: 2, payload sum: 4 bytes
      Outgoing packet: CHAT_FILTER_SETTINGS_PRIVATECHAT, count: 2, payload sum: 2 bytes
      Outgoing packet: SERVER_TICK_END, count: 522, payload sum: 0 bytes
      Outgoing packet: VARP_RESET, count: 4, payload sum: 0 bytes
      Outgoing packet: CAM_RESET, count: 4, payload sum: 0 bytes
      Outgoing packet: CLEAR_ENTITIES, count: 2, payload sum: 0 bytes
      Outgoing packet: FRIENDLIST_LOADED, count: 2, payload sum: 0 bytes
      Outgoing packet: UPDATE_FRIENDLIST, count: 2, payload sum: 0 bytes
      Disconnection reason: LOGOUT, count: 1

Login Blocks
  Inet Address: /127.0.0.1
    Login Block
      Version: 225
      Sub Version: 1
      Client Type: DESKTOP
      Platform Type: DEFAULT
      Seed: [-475513568, -906498712, -787741855, -219195878]
      Session Id: -691,196,275,561,316,263
      Username: Kris
      Low Detail: false
      Resizable: true
      Width: 1126
      Height: 759
      UUID: [-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1]
      Site Settings: p1cxIGuofoI*MdvgTjPbwNT68nYEsLxmwYfDOqPAxNr9FhEej0165g
      Affiliate: 0
      CRC Header: 0
      CRC: [1402716299, -1157264799, -1059355604, 1681681460, -910027546, -669582295, 817340927, -1044285571, -1444034203, -1467756878, 737620165, 933155772, 1621685459, -1207822887, 2006043712, 1341590283, 0, -919261486, -642006582, 1152447416, 1705274935]
      Host Platform Stats
        Version: 9
        OS Type: 1
        OS 64 Bit: true
        OS Version: 11
        Java Vendor: 2
        Java: 2 17.0.10
        Applet: true
        Java Max Memory (MB): 228
        Java Available Processors: 32
        System Memory: 0
        System Speed: 0
        GPU DX Name: 
        GPU GL Name: 
        GPU GL Version: 
        GPU Driver Date: 0.0
        CPU Manufacturer: 
        CPU Brand: 
        CPU Counts: 0, 0
        CPU Features: [0, 0, 0]
        CPU Signature: 0
        Client Name: 
        Device Name: 
    Login Block
      Version: 225
      Sub Version: 1
      Client Type: DESKTOP
      Platform Type: DEFAULT
      Seed: [1544922788, -734024476, 864758687, -71580636]
      Session Id: 2,473,695,465,378,212,025
      Username: Kris
      Low Detail: false
      Resizable: true
      Width: 1126
      Height: 759
      UUID: [-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1]
      Site Settings: p1cxIGuofoI*MdvgTjPbwNT68nYEsLxmwYfDOqPAxNr9FhEej0165g
      Affiliate: 0
      CRC Header: 0
      CRC: [1402716299, -1157264799, -1059355604, 1681681460, -910027546, -669582295, 817340927, -1044285571, -1444034203, -1467756878, 737620165, 933155772, 1621685459, -1207822887, 2006043712, 1341590283, 0, -919261486, -642006582, 1152447416, 1705274935]
      Host Platform Stats
        Version: 9
        OS Type: 1
        OS 64 Bit: true
        OS Version: 11
        Java Vendor: 2
        Java: 2 17.0.10
        Applet: true
        Java Max Memory (MB): 228
        Java Available Processors: 32
        System Memory: 0
        System Speed: 0
        GPU DX Name: 
        GPU GL Name: 
        GPU GL Version: 
        GPU Driver Date: 0.0
        CPU Manufacturer: 
        CPU Brand: 
        CPU Counts: 0, 0
        CPU Features: [0, 0, 0]
        CPU Signature: 0
        Client Name: 
        Device Name: 
    Login Block
      Version: 225
      Sub Version: 1
      Client Type: ENHANCED_WINDOWS
      Platform Type: DEFAULT
      Seed: [-1203597627, 5973525, -2086316526, 1361944760]
      Session Id: -5,581,999,782,346,983,246
      Username: Kris@hi
      Low Detail: false
      Resizable: true
      Width: 993
      Height: 716
      UUID: [2, -61, -81, -38, -117, -63, 69, 38, -36, -19, 27, 14, -34, 118, 0, -101, 63, -25, 77, -53, 7, -37, 2, -125]
      Site Settings: p1cxIGuofoI*MdvgTjPbwNT68nYEsLxmwYfDOqPAxNr9FhEej0165g
      Affiliate: 0
      CRC Header: 249
      CRC: [1402716299, -1157264799, -1059355604, 1681681460, -910027546, -669582295, 817340927, -1044285571, -1444034203, -1467756878, 737620165, 933155772, 1621685459, -1207822887, 2006043712, 1341590283, 0, -919261486, -642006582, 1152447416, 1705274935]
      Host Platform Stats
        Version: 9
        OS Type: 1
        OS 64 Bit: true
        OS Version: 11
        Java Vendor: 0
        Java: 0 0.0.0
        Applet: true
        Java Max Memory (MB): 0
        Java Available Processors: 1
        System Memory: 65444
        System Speed: 0
        GPU DX Name: 
        GPU GL Name: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
        GPU GL Version: 4.6.0 NVIDIA 561.09
        GPU Driver Date: 9.2024
        CPU Manufacturer: AuthenticAMD
        CPU Brand: AMD Ryzen 9 3950X 16-Core Processor
        CPU Counts: 32, 32
        CPU Features: [-19385845, 395049983, 802421759]
        CPU Signature: 8851216
        Client Name: NXT-Windows-64
        Device Name: 
    Login Block
      Version: 225
      Sub Version: 1
      Client Type: ENHANCED_WINDOWS
      Platform Type: DEFAULT
      Seed: [633418785, 1581645915, 227515383, -1830978077]
      Session Id: -2,724,693,611,211,500,808
      Username: Kris lololollollolololol
      Low Detail: false
      Resizable: true
      Width: 993
      Height: 716
      UUID: [2, -61, -81, -38, -117, -63, 69, 38, -36, -19, 27, 14, -34, 118, 0, -101, 63, -25, 77, -53, 7, -37, 2, -125]
      Site Settings: p1cxIGuofoI*MdvgTjPbwNT68nYEsLxmwYfDOqPAxNr9FhEej0165g
      Affiliate: 0
      CRC Header: 249
      CRC: [1402716299, -1157264799, -1059355604, 1681681460, -910027546, -669582295, 817340927, -1044285571, -1444034203, -1467756878, 737620165, 933155772, 1621685459, -1207822887, 2006043712, 1341590283, 0, -919261486, -642006582, 1152447416, 1705274935]
      Host Platform Stats
        Version: 9
        OS Type: 1
        OS 64 Bit: true
        OS Version: 11
        Java Vendor: 0
        Java: 0 0.0.0
        Applet: true
        Java Max Memory (MB): 0
        Java Available Processors: 1
        System Memory: 65444
        System Speed: 0
        GPU DX Name: 
        GPU GL Name: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
        GPU GL Version: 4.6.0 NVIDIA 561.09
        GPU Driver Date: 9.2024
        CPU Manufacturer: AuthenticAMD
        CPU Brand: AMD Ryzen 9 3950X 16-Core Processor
        CPU Counts: 32, 32
        CPU Features: [-19385845, 395049983, 802421759]
        CPU Signature: 8851216
        Client Name: NXT-Windows-64
        Device Name: 
    Login Block
      Version: 225
      Sub Version: 1
      Client Type: ENHANCED_WINDOWS
      Platform Type: DEFAULT
      Seed: [1620829800, -1176032415, 2023924781, -2004085757]
      Session Id: 1,157,503,768,099,201,687
      Username: Kris
      Low Detail: false
      Resizable: true
      Width: 993
      Height: 716
      UUID: [2, -61, -81, -38, -117, -63, 69, 38, -36, -19, 27, 14, -34, 118, 0, -101, 63, -25, 77, -53, 7, -37, 2, -125]
      Site Settings: p1cxIGuofoI*MdvgTjPbwNT68nYEsLxmwYfDOqPAxNr9FhEej0165g
      Affiliate: 0
      CRC Header: 249
      CRC: [1402716299, -1157264799, -1059355604, 1681681460, -910027546, -669582295, 817340927, -1044285571, -1444034203, -1467756878, 737620165, 933155772, 1621685459, -1207822887, 2006043712, 1341590283, 0, -919261486, -642006582, 1152447416, 1705274935]
      Host Platform Stats
        Version: 9
        OS Type: 1
        OS 64 Bit: true
        OS Version: 11
        Java Vendor: 0
        Java: 0 0.0.0
        Applet: true
        Java Max Memory (MB): 0
        Java Available Processors: 1
        System Memory: 65444
        System Speed: 0
        GPU DX Name: 
        GPU GL Name: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
        GPU GL Version: 4.6.0 NVIDIA 561.09
        GPU Driver Date: 9.2024
        CPU Manufacturer: AuthenticAMD
        CPU Brand: AMD Ryzen 9 3950X 16-Core Processor
        CPU Counts: 32, 32
        CPU Features: [-19385845, 395049983, 802421759]
        CPU Signature: 8851216
        Client Name: NXT-Windows-64
        Device Name: 
```

</details>

### CODE REVIEW

File: protocol/src/main/kotlin/net/rsprot/protocol/metrics/impl/GenericNetworkTrafficHandler.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Queue
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicInteger
+
+public class GenericNetworkTrafficHandler<LoginBlock>(
+    private val lock: TrafficHandlerLock,
+    override var loginChannelTrafficHandler: LoginChannelTrafficHandler,
+    override var js5ChannelTrafficHandler: Js5ChannelTrafficHandler,
+    override var gameChannelTrafficHandler: GameChannelTrafficHandler,
```
Comment (Dill-Lan) [inline protocol/src/main/kotlin/net/rsprot/protocol/metrics/impl/GenericNetworkTrafficHandler.kt:21]: seems to me like these can be `val`s unless i'm missing some mutation somewhere

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import net.rsprot.protocol.metrics.snapshots.impl.GenericNetworkTrafficSnapshot
+import net.rsprot.protocol.metrics.writer.NetworkTrafficWriter
+import java.net.InetAddress
+import java.text.NumberFormat
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import kotlin.collections.iterator
+
+public data object GenericNetworkTrafficWriter : NetworkTrafficWriter<GenericNetworkTrafficSnapshot<*>, String> {
```
Comment (Dill-Lan) [inline protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:16]: i feel like there's a good case for building another layer on top of this where consumers can set up a regular writing of traffic snapshot contents to a named file in some folder

also perhaps this is better named as a `StringNetworkTrafficWriter`, where there'd also be a `FileNetworkTrafficWriter` that delegates the formatting to this, but has that file-writing-specific code encapsulated within it

what to you think about extending the api a bit for this type of thing? i can give that a crack if you think it'd be valuable. just conscious that all your consumers might end up writing the same thing across the diff projects

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    private const val INDENT: String = "  "
```
Comment (Dill-Lan) [inline protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:17]: y u no `"\t"`

Comment (Dill-Lan) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:17]: .... maybe that's a controversial topic 😂

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt
Comment (Dill-Lan) [inline protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:1]: nit: i kinda don't like the `Generic` prefix for these classes. to me, it feels like these are more of a 'default' implementation, but not 'generic' - which has other specific meanings in programming ya know?

Comment (Z-Kris) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:1]: Already renamed in an unpushed commit. These implementations are called Concurrent now, as the underlying implementation utilizes concurrent collections and atomic/volatile properties.

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt (hunk 3 above)
Comment (Z-Kris) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:17]: Tabs bad. Two spaces great. End of conversation!

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt (hunk 2 above)
Comment (Z-Kris) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:16]: This is the bit I'm not really sure about in general. There's a lot of logging happening and I feel like most people will not want like half of it, especially when it comes to all the details within the login block.
My thought process here was to make a default implementation that servers can use to get a rough idea of what gets tracked and how, but when it came to actually making use of it, they'd write their own.
Perhaps there is a way to make a nicer API for it, I haven't really figured out anything just yet.

If you want, go ahead and take a crack at it. Do so after I'm done with this PR and have merged it into the master branch, though. That will be when I stop actively updating this feature (it's fine to update it after it's gone to master, this won't be published until after I finish all the things I wanted to do for your server anyhow).

File: protocol/src/main/kotlin/net/rsprot/protocol/metrics/impl/GenericNetworkTrafficHandler.kt (hunk 1 above)
Comment (Z-Kris) [reply protocol/src/main/kotlin/net/rsprot/protocol/metrics/impl/GenericNetworkTrafficHandler.kt:21]: You are correct, yeah. I initially intended to swap out the instances of that whenever a snapshot was created, but that did not quite work out, need to reset internally instead. I've changed it in a future commit.

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt
Comment (Dill-Lan) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:1]: brilliant

File: protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt (hunk 2 above)
Comment (Dill-Lan) [reply protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt:16]: fair nuff - i guess it's better to see how this pans out in practice, and make changes that solve real problems. im just making stuff up atm

sounds good to me!

**Pull request merged**